src/pyseekdb/client/client_base.py

```
@@ -706,39 +718,98 @@ def create_collection(
+            if embedding_function is not None and EmbeddingFunction.support_persistence(embedding_function):
+                settings["embedding_function"] = {
+                    "name": embedding_function.name(),
+                    "properties": embedding_function.get_config()
+                }
 
-        Returns:
-            Collection object
+            settings_str = json.dumps(settings)
+
+            self._create_sdk_collections_if_not_exists()
+            delete_sql = f"DELETE FROM {CollectionNames.sdk_collections_table_name()} WHERE COLLECTION_NAME = '{collection_name}'"
```

Ruff (0.14.13)

Possible SQL injection vector through string-based query construction

(S608)